controllers/dspastatus/dspa_status.go

```diff
@@ -0,0 +1,191 @@
+package dspastatus
+
+import (
+	"fmt"
+	dspav1alpha1 "github.com/opendatahub-io/data-science-pipelines-operator/api/v1alpha1"
+	"github.com/opendatahub-io/data-science-pipelines-operator/controllers/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type DSPAStatus interface {
+	SetDatabaseReady()
+	SetDatabaseNotReady(err error)
+
+	SetObjStoreReady()
+	SetObjStoreNotReady(err error)
+
+	SetApiServerStatus(apiServerReady metav1.Condition)
+
+	SetPersistenceAgentStatus(persistenceAgentReady metav1.Condition)
+
+	SetScheduledWorkflowStatus(scheduledWorkflowReady metav1.Condition)
```

Comment:
It feels like you are doing a lot of work to satisfy "Ready" and "NotReady", [[1]](https://github.com/hbelmiro/data-science-pipelines-operator/blob/RHOAIENG-6287/controllers/dspipeline_controller.go#L340-L353), [[2]](https://github.com/hbelmiro/data-science-pipelines-operator/blob/RHOAIENG-6287/controllers/dspastatus/dspa_status.go#L52-L83)

Do you think it would be easier if we just had: 

SetComponentCondition(Type, Status, Reason, Err) 

Where "Component" = (APIServer, SWF, DB, ObjStore, PA) 
And Err is optional
?

Reply:
It would be easier.
But usually, I prefer to have more meaningful functions to increase readability and prevent the caller from knowing too much about implementation details.

For example:

```go
dspaStatus.SetDatabaseReady()
```

Is easier to read than:

```go
SetComponentCondition(config.DatabaseAvailable, metav1.ConditionTrue, config.DatabaseAvailable, nil)
```

I'm fine with changing if you think it will be better. Just let me know. It's a quick change.

Reply:
Hmm I would not consider "Condition Type" and "Condition Reason" an implementation detail for setting a component "Ready" unless the method was named "SetDatabaseAvailabilityReady"

Anyways, I don't feel very strongly about this, so it's up to you. Resolving this.